test/Tests/src/Semantic/Meta_Spec.enso

```diff
@@ -437,4 +437,14 @@ spec =
             Meta.type_of (_.is_nothing) . should_equal Function
             Meta.type_of (.is_nothing) . should_equal Function
 
+    Test.group "Meta.Type.find"
+        Test.specify "find boolean and use it" <|
+            fqn = "Standard.Base.Data.Boolean.Boolean"
+            typ = Meta.Type.find fqn
+            typ . is_a Meta.Type . should_be_true
+            typ.name . should_equal "Boolean"
+            #!!! typ.qualified_name . should_equal fqn
```

Comment:
However there are problems. When you search for a `Boolean` by its FQN you get a type which **has different FQN**! That's caused by implementation of [Type.getQualifiedName](https://github.com/enso-org/enso/blob/df4183e78e914a60b8503bd67d996d26b2ae7fac/engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java#L88) which checks for `getAssociatedType` and then behaves differently.

What shall call to `Meta.Type.find "Standard.Base.Data.Boolean"` do? Options:
- find the module
- find the `getAssociatedType` of the module
- yield an error

Reply:
What is the value of `typ.qualified_name`?

Reply:
> I think it should return the associated type

But then the `Meta.Type.find "Standard.Base.Data.Boolean"` will behave differently than `import Standard.Base.Data.Boolean` - isn't that also a problem?

Reply:
> > I think it should return the associated type
> 
> But then the `Meta.Type.find "Standard.Base.Data.Boolean"` will behave differently than `import Standard.Base.Data.Boolean` - isn't that also a problem?

Are you sure?
```patch
diff --git a/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/debug/DebugEvalNode.java b/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/debug/DebugEvalNode.java
index bcf9a54e8..b2d263469 100644
--- a/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/debug/DebugEvalNode.java
+++ b/engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/debug/DebugEvalNode.java
@@ -2,10 +2,12 @@ package org.enso.interpreter.node.expression.builtin.debug;
 
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.nodes.Node;
+import java.util.Arrays;
 import org.enso.interpreter.dsl.BuiltinMethod;
 import org.enso.interpreter.node.BaseNode;
 import org.enso.interpreter.node.expression.builtin.text.util.ExpectTextNode;
 import org.enso.interpreter.node.expression.debug.EvalNode;
+import org.enso.interpreter.runtime.EnsoContext;
 import org.enso.interpreter.runtime.callable.CallerInfo;
 import org.enso.interpreter.runtime.state.State;
 
@@ -25,6 +27,12 @@ public class DebugEvalNode extends Node {
 
   Object execute(
       VirtualFrame requestOwnStackFrame, CallerInfo callerInfo, State state, Object expression) {
-    return evalNode.execute(callerInfo, state, expectTextNode.execute(expression));
+    var ctx = EnsoContext.get(this);
+    var module = ctx.getPackageRepository().getLoadedModule("Standard.Base.Data.Boolean").get();
+    var moduleScope = module.getScope();
+    var associatedType = moduleScope.getAssociatedType();
+    var methodNames = moduleScope.getMethodNamesForType(associatedType);
+    System.out.println("Methods for associated type of module 'Standard.Base.Data.Boolean': " + methodNames);
+    return ctx.getNothing();
   }
 }
```

And `Debug.eval ""` prints
```
Methods for associated type of module 'Standard.Base.Data.Boolean': [True, False, Boolean]
```

So they look the same?

Reply:
Returning associated type since cb5f01230bdf7ca18078822e778b46bb5272534b